README.md

```diff
@@ -8,6 +8,265 @@
 
 ## Overview
 
+In the previous lessons, we learned how to communicate between components by
+passing around handler functions, e.g. our popover had an `onClose` prop or
+an input might have an `onChange` prop.
+
+While it's certainly possible to structure your component hierarchy using
+`on...` handlers, this approach is rather inflexible and leads to a lot of code
+duplication in the long term.
+
+In this lesson we're going to be learning how decoupling those functions into
```

Comment:
We try to make lessons feel super decoupled — meaning we don't reference to any 'previous lessons' (even though it's  intentionally vague here). I think we can ditch this entire leading paragraph.